### DISCUSSION TITLE
Repository verification function

### DISCUSSION THREAD

**Discussion by gtdang** [AutoResearch/autora#399] [Ideas]
Younes and I were brainstorming on this idea. 
Create an application that will check repository consistency and compatibility. 

Checks for: 
* directory structure is up-to-date with current template 
* naming conventions
* files: mkdocs, pyproject.toml, files in the docs directory
* init files are where they should be 
* run and verify tests 

Where could this run?
* Github action in autora mother package  
* CLI app 

How often would it run? 
* Every time autora mother package is updated?  
* Any time the temple is updated?

**Comment by benwandrew** [comment]
so like subpackage hooks? to start, probably best to put the onus on us for resolving any issues that are surfaced (as opposed to frustrating a contributor who is trying to create a subpackage), so maybe every time mother package is updated? could be part of our internal vetting process for approving integration of the new subpackage with mother package

**Comment by musslick** [comment]
Fantastic idea!
